### PR TITLE
[Devops] Move to use provisionator to get xcode.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Move to use the provisionator and add a make file to generat the script
so that we only have to bump xcode in a single place.

Move all deps to provisionator and make provisionator download and
install the xi pkg that has been built to be tested in the pipeline, we
do not longer use curl to do it.

### CODE REVIEW

File: tools/devops/xcode-deps.csx.in [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Item ("https://xamjenkinsartifact.azureedge.net/build-package-osx-mono/2020-02/76/d6f0c25d34da7512639346828ef368ff65a25003/MonoFramework-MDK-6.12.0.73.macos10.xamarin.universal.pkg", kind: ItemDependencyKind.AtLeast);
+Item ("https://dl.xamarin.com/VsMac/VisualStudioForMac-8.5.5.7.dmg");
+Item ("@XI_PACKAGE@");
+
+BrewPackage ("p7zip");
+
+void BrewPackage (string name)
+{
+	// Assumes brew is already installed. 
+	// All Macs provisioned by Xamarin, VSEng, or DDFUN should have brew by default!
```
Comment (whitneyschmidt) [inline tools/devops/xcode-deps.csx.in:17]: Are there any other dependencies that need to be taken into account here?

Comment (mandel-macaque) [reply tools/devops/xcode-deps.csx.in:21]: AFAIK, nope

File: tools/devops/xcode-deps.csx.in [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#r "_provisionator/provisionator.dll"
+
+using System.IO;
+using System.Linq;
+
+using static Xamarin.Provisioning.ProvisioningScript;
+
+Exec ("defaults", "write", "com.apple.LaunchServices", "LSQuarantine", "-bool", "NO");
```
Comment (spouliot) [inline tools/devops/xcode-deps.csx.in:8]: that's a global setting

File: tools/devops/xcode-deps.csx.in [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+Xcode ("@XCODE_VERSION@").XcodeSelect ();
+
+// provisionator knows how to deal with this items
+Item ("https://xamjenkinsartifact.azureedge.net/build-package-osx-mono/2020-02/76/d6f0c25d34da7512639346828ef368ff65a25003/MonoFramework-MDK-6.12.0.73.macos10.xamarin.universal.pkg", kind: ItemDependencyKind.AtLeast);
```
Comment (spouliot) [inline tools/devops/xcode-deps.csx.in:12]: can you use the URL from Make.config ?

File: tools/devops/xcode-deps.csx.in [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+Item ("https://dl.xamarin.com/VsMac/VisualStudioForMac-8.5.5.7.dmg");
```
Comment (spouliot) [inline tools/devops/xcode-deps.csx.in:13]: same, can you use the URL from Make.config ?

File: tools/devops/xcode-deps.csx.in (hunk 2 above)
Comment (mandel-macaque) [reply tools/devops/xcode-deps.csx.in:8]: This is the only part i'm not happy with, but we need to set the attrs somehow then

Comment (spouliot) [reply tools/devops/xcode-deps.csx.in]: you can do this to specific files using `xattr` like @dalexsoto mentioned it recently in #macios-tnt

File: .gitignore [hunk 5]
```diff
@@ -32,3 +32,4 @@ tests/bcl-test/SystemCoreXunit*.csproj
 tests/bcl-test/SystemXunit.csproj
 global.json
 .idea
+xcode-deps.csx
```
Comment (rolfbjarne) [inline .gitignore:35]: Missing eol.

File: tools/devops/xcode-deps.csx.in [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+using System.IO;
+using System.Linq;
+
+using static Xamarin.Provisioning.ProvisioningScript;
+
+Xcode ("@XCODE_VERSION@").XcodeSelect (allowUntrusted: true);
+
+// provisionator knows how to deal with this items
+Item ("@MONO_PACKAGE@", kind: ItemDependencyKind.AtLeast);
```
Comment (rolfbjarne) [inline tools/devops/xcode-deps.csx.in:11]: I'm not sure the `AtLeast` part is right, I think we currently install this exact package, which means we'll want the exact version.

File: tools/devops/Makefile [hunk 7]
```diff
@@ -0,0 +1,10 @@
+TOP=../..
+include $(TOP)/Make.config
+
+xcode-deps.csx: xcode-deps.csx.in Makefile $(TOP)/Make.config
```
Comment (rolfbjarne) [inline tools/devops/Makefile:4]: You're not only installing Xcode, but several other dependencies as well, which makes the filename misleading. What about something like `device-tests.csx` instead?

Comment (dalexsoto) [reply tools/devops/Makefile:4]: `device-tests-provisioning.csx` Would make it a little more explicit the intent

**Comment by mandel-macaque** [comment]
@monojenkins backport d16-7

**Pull request merged**